uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/models/instances/leave-request-instance.js

```diff
@@ -73,12 +78,12 @@ define([
                   self.comments[index] = commentData;
                 }));
             })(index);
-          } else if(comment.toBeDeleted) {
+          } else if (comment.toBeDeleted) {
             promises.push(LeaveRequestAPI.deleteComment(comment.comment_id));
           }
         });
-
-        return $q.all(promises);
+        
+        return promises.length ? $q.all(promises) : $q.resolve();
```

Comment:
now restored it and test cases pass, earlier they were breaking so had to put this check